core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1405,39 +1189,36 @@ object GroupMetadataManager {
       null
     } else {
       val version = buffer.getShort
-      val valueSchema = schemaForGroupValue(version)
-      val value = valueSchema.read(buffer)
-
       if (version >= 0 && version <= CURRENT_GROUP_METADATA_VALUE_SCHEMA_VERSION) {
-        val generationId = value.get(GENERATION_KEY).asInstanceOf[Int]
-        val protocolType = value.get(PROTOCOL_TYPE_KEY).asInstanceOf[String]
-        val protocol = value.get(PROTOCOL_KEY).asInstanceOf[String]
-        val leaderId = value.get(LEADER_KEY).asInstanceOf[String]
-        val memberMetadataArray = value.getArray(MEMBERS_KEY)
+        val value = new GroupMetadataValue(new ByteBufferAccessor(buffer), version)
+        val generationId = value.generation
+        val protocolType = value.protocolType
+        val protocol = value.protocol
+        val leaderId = value.leader
+        val memberMetadataArray = value.members.asScala
         val initialState = if (memberMetadataArray.isEmpty) Empty else Stable
         val currentStateTimestamp: Option[Long] =
-          if (version >= 2 && value.hasField(CURRENT_STATE_TIMESTAMP_KEY)) {
-            val timestamp = value.getLong(CURRENT_STATE_TIMESTAMP_KEY)
+          if (version >= 2) {
+            val timestamp = value.currentStateTimestamp
```

Comment:
I am not sure if this is worth it but we could avoid checking the version by putting `-1` as default value for `currentStateTimestamp` in the schema.